### PR TITLE
Add Area narrative and fix key/slug clobbering bug in Area admin

### LINKED ISSUE #119: add support for community area narrative
1. add a narrative attribute to the `Area` model, and allow editing of this attribute in the admin interface.
2. display the narrative in a popover to the map ui, triggerable by one or more links (e.g., in the top-right info box on the map, perhaps in the sidebar).

### PR DESCRIPTION
**Pull request by jeancochrane**
## Overview

* Add `Area.narrative` field and display it with the community area name in the sidebar
* Change automatic generation of key/slug fields for `Area` models so that they only get generated when the model is first saved

Closes #119.

## Notes

I'm not super confident in the placement of the community area names and narratives. #119 suggested that they be added as a toggleable popover to the map, but I'm not sure how to do that without taking up too much map space.

## Testing instructions

* Run migrations with `python manage.py migrate`
* Run the new tests with `python manage.py test`
* Run a dev server with `python manage.py runserver`
* Navigate to http://localhost:8000/admin and log in with your admin user
* Navigate to `Partner communities > Kankakee County` and save a `Short description`
* Navigate to http://localhost:8000/map/?area=kankakee and view your short description

### CODE REVIEW

File: edi/admin.py [hunk 1]
```diff
@@ -55,9 +55,10 @@ def save_slug_and_key(self, obj):
 
     def save_model(self, request, obj, form, change):
         '''
-        Autopopulate slug and key fields.
+        Autopopulate slug and key fields for new objects.
         '''
-        self.save_slug_and_key(obj)
+        if not change:
+            self.save_slug_and_key(obj)
```
Comment (jeancochrane) [inline edi/admin.py]: Without this change, the slug/key gets updated every time the model is saved. This is particularly bad for `Kankakee County`, which uses the key/slug `kankakee` for historical reasons but would have the slug `kankakee-county` based on the rules in `save_slug_and_key`. We really only need the key/slug to be set when the object is created.

File: edi/migrations/0026_auto_20190311_2116.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('edi', '0025_auto_20180514_2012'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='overlay',
+            name='category',
+            field=models.ForeignKey(help_text='Select a category that describes this overlay so that we can group it into a dropdown menu.', on_delete=django.db.models.deletion.CASCADE, to='edi.OverlayCategory'),
```
Comment (jeancochrane) [inline edi/migrations/0026_auto_20190311_2116.py]: This migration got generated when I ran `makemigrations` without changing anything in this PR. I think it's a leftover migration from a previous change to some help text.

File: edi/tests.py [hunk 3]
```diff
@@ -1,3 +1,51 @@
 from django.test import TestCase
+from django.contrib.admin.sites import AdminSite
 
-# Create your tests here.
+from edi import models, admin, forms
+
+
+class TestAdmin(TestCase):
```
[the remark this answers is not in the document]
Comment (jeancochrane) [reply edi/tests.py]: Oops, forgot about pytest. Thanks for the reminder! This should be an easy change.

**Comment by jeancochrane** [comment]
Requesting @derekeder's review to help shed some light on the narrative placement.

**Comment by jeancochrane** [comment]
Gotcha, thanks @derekeder!

**Comment by jeancochrane** [comment]
I made a quick edit to turn this into a modal. Here's what the page looks like with modal not displayed:

edi-read-more

And here's what it looks like when you open the modal:

edi-read-more-modal

**Comment by derekeder** [comment]
@jeancochrane these changes look good to me! It looks like @hancush also needs to approve.

**Pull request merged**